platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ResolvedArtifactCollectingVisitor.java

```diff
@@ -60,10 +64,37 @@ public boolean requireArtifactFiles() {
     }
 
     public Set<ResolvedArtifactResult> getArtifacts() {
-        return artifacts;
+        return Sets.newLinkedHashSet(artifacts.values());
```

Comment:
❓ Shouldn't this just use `HashSet`?  

I don't think it's possible for this to preserve the same order it previously returned, which was based upon the order that the values (`ResolvedArtifactResult`s) were encountered during iteration in `visitArtifact`.  Now the order will be based on the order that _unique keys_ (`ArtifactIdentifier`s) are encountered.  This could be a meaningful difference.  But we're only promising to return a `Set` here, so why try and preserve an ordering that isn't really well-defined anyways?